eng/testing/tests.mobile.targets

```diff
@@ -148,26 +148,22 @@
 
   </Target>
 
-  <Target Name="ConfigureTrimming" Condition="'$(EnableAggressiveTrimming)' == 'true'" BeforeTargets="PrepareForILLink">
+  <Target Name="ConfigureTrimming" Condition="'$(EnableAggressiveTrimming)' == 'true'" AfterTargets="AddTestRunnersToPublishedFiles">
     <PropertyGroup>
       <TrimMode>link</TrimMode>
     </PropertyGroup>
 
     <ItemGroup>
-      <ManagedAssemblyToLink Include="$(OutDir)\*xunit*">
-        <IsTrimmable>true</IsTrimmable>
-        <TrimMode>link</TrimMode>
-      </ManagedAssemblyToLink>
-      <ManagedAssemblyToLink Condition="('%(ManagedAssemblyToLink.FileName).dll' != '$(MSBuildProjectName).dll')" >
-        <TrimMode>link</TrimMode>
-      </ManagedAssemblyToLink>
-      <ManagedAssemblyToLink Condition="('%(ManagedAssemblyToLink.FileName).dll' == '$(AssemblyName).dll')" >
-        <TrimMode>copy</TrimMode>
-      </ManagedAssemblyToLink>
-      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'Android'" Include="AndroidTestRunner"/>
-      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'Browser'" Include="WasmTestRunner"/>
-      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'iOS'" Include="AppleTestRunner"/>
-      <TrimmerRootAssembly Include="$(AssemblyName)"/>
+      <!-- Mark all the assemblies for link. We will explicitly mark the non-trimmable ones -->
+      <ResolvedFileToPublish TrimMode="link" />
+
+      <!-- Don't trim the test runner, and main assembly.
```

Comment:
We want to trim the runner just not the test assembly

Reply:
ah ok. We were earlier passing `-a WasmTestRunner` to `illink.dll` - does that not mean that we shouldn't be trimming it?

Reply:
With the changes here we get:

```
                    -a "/Users/radical/dev/r2/artifacts/bin/WasmTestRunner/net6.0-Release/WasmTestRunner.dll"
                     -a "/Users/radical/dev/r2/artifacts/obj/System.Buffers.Tests/net6.0-Release/browser-wasm/System.Buffers.Tests.dll"

                     -reference "/Users/radical/dev/r2/artifacts/bin/WasmTestRunner/net6.0-Release/WasmTestRunner.dll"
                     -p copy "WasmTestRunner"
                     -reference "/Users/radical/dev/r2/artifacts/obj/System.Buffers.Tests/net6.0-Release/browser-wasm/System.Buffers.Tests.dll"
                     -p copy "System.Buffers.Tests"
```

.. instead we want only the test assembly to be `copy`. But what should the `-a` args be?

Reply:
-a args are correct, you need to tweak `-p copy "WasmTestRunner"` to be `-p link "WasmTestRunner"` though